recipes/hive.rb

```diff
@@ -24,7 +24,13 @@
 end
 
 hive_conf_dir = "/etc/hive/#{node['hive']['conf_dir']}"
-hive_data_dir = '/usr/lib/hive/lib'
+
+if node['hadoop']['distribution'] == 'hdp' && node['hadoop']['distribution_version'].to_f == 2.2
+  hive_data_dir = '/usr/hdp/current/hive-server2/lib'
+else
+  hive_data_dir = '/usr/lib/hive/lib'
+end
```

Comment:
-rwxr-xr-x 1 root root 4535 Apr 29 22:43 /etc/init.d/hive-server2

Here are my choices for pointing to the right linked lib directory for hive under hdp 2.2:

root@hdp22-ubuntu-1204:/usr/hdp/current# ls -ld /usr/hdp/current/hive*
lrwxrwxrwx 1 root root 23 Apr 29 22:41 /usr/hdp/current/hive-client -> /usr/hdp/2.2.4.2-2/hive
lrwxrwxrwx 1 root root 23 Apr 29 22:43 /usr/hdp/current/hive-metastore -> /usr/hdp/2.2.4.2-2/hive
lrwxrwxrwx 1 root root 23 Apr 29 22:43 /usr/hdp/current/hive-server2 -> /usr/hdp/2.2.4.2-2/hive

hive-server2 would not be my first choice if there were a /usr/hdp/current/hive link.